druid/src/widget/list.rs

```diff
@@ -71,6 +74,53 @@ pub trait ListIter<T>: Data {
     fn data_len(&self) -> usize;
 }
 
+#[cfg(feature = "im")]
+impl<T: Data> ListIter<T> for Vector<T> {
+    fn for_each(&self, mut cb: impl FnMut(&T, usize)) {
+        for (i, item) in self.iter().enumerate() {
+            cb(item, i);
+        }
+    }
+
+    fn for_each_mut(&mut self, mut cb: impl FnMut(&mut T, usize)) {
+        for (i, item) in self.iter_mut().enumerate() {
+            cb(item, i);
+        }
+    }
+
+    fn data_len(&self) -> usize {
+        self.len()
+    }
+}
+
+#[cfg(feature = "im")]
+impl<T1: Data, T: Data> ListIter<(T1, T)> for (T1, Vector<T>) {
```

Comment:
Maybe call them `Shared` and `T` instead? Or at least `S` and `T`.
I still do not understand why people like useless names for generic parameters,
this is really not something where we should let mathematicians inspire us.

Reply:
I just copied that from the `Arc<Vec>` implementation, but yeah it took me a while to grasp what the `T1` even is, because I hadn't used `List` before. I changed it to `S` and added a comment.

Reply:
Thanks! At first I got it the wrong way, thought `T1` would be the iterable due to the `1`.